packages/editor/src/browser/editor-quick-open-service.ts

```diff
@@ -0,0 +1,137 @@
+
+    @inject(OpenerService)
+    protected readonly openerService: OpenerService;
+
+    @inject(PrefixQuickOpenService)
+    protected readonly prefixQuickOpenService: PrefixQuickOpenService;
+
+    @inject(LabelProvider)
+    protected readonly labelProvider: LabelProvider;
+
+    @inject(EditorManager)
+    protected readonly editorManager: EditorManager;
+
+    readonly prefix: string = 'edt ';
+
+    get description(): string {
+        return 'Show All Opened Editors';
+    }
+
+    getModel(): QuickOpenModel {
+        return this;
+    }
+
+    getOptions(): QuickOpenOptions {
+        return {
+            fuzzyMatchLabel: {
+                enableSeparateSubstringMatching: true
+            },
+            fuzzyMatchDescription: {
+                enableSeparateSubstringMatching: true
+            },
+            onClose: () => {
+                this.cancelIndicator.cancel();
+            }
+        };
+    }
+
+    open(): void {
+        this.prefixQuickOpenService.open(this.prefix);
+    }
+
+    private cancelIndicator = new CancellationTokenSource();
+
+    public async onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): Promise<void> {
+        this.cancelIndicator.cancel();
+        this.cancelIndicator = new CancellationTokenSource();
+        const token = this.cancelIndicator.token;
+
+        const editorItems: QuickOpenItem[] = [];
+
+        // Get the alphabetically sorted list of URIs of all currently opened Editor widgets.
+        const widgets: URI[] = this.editorManager.all
+            .filter((w: EditorWidget) => !!w.getResourceUri())
+            .map((w: EditorWidget) => w.getResourceUri()!)
+            .sort();
+
+        if (widgets.length === 0) {
+            editorItems.push(new QuickOpenItem({
+                label: 'List of opened editors is currently empty',
+                run: () => false
+            }));
+            acceptor(editorItems);
+            return;
+        }
+
+        for (const uri of widgets) {
+            const item = await this.toItem(uri);
+            if (!!token.isCancellationRequested) {
+                return;
+            }
+            editorItems.push(item);
+            acceptor(editorItems);
+        }
+        return;
+    }
+
+    protected async toItem(uri: URI, group?: QuickOpenGroupItemOptions): Promise<QuickOpenItem<QuickOpenItemOptions>> {
+        const description = this.labelProvider.getLongName(uri.parent);
+
+        const options: QuickOpenItemOptions = {
+            label: this.labelProvider.getName(uri),
+            iconClass: await this.labelProvider.getIcon(uri) + ' file-icon',
+            description: description,
+            tooltip: uri.path.toString(),
+            uri: uri,
+            hidden: false,
+            run: this.getRunFunction(uri)
+        };
+        return new QuickOpenItem<QuickOpenItemOptions>(options);
+    }
+
+    // Calls the openFile function to open file of the given URI.
```

Comment:
Please follow other documentation for methods, example:

https://github.com/theia-ide/theia/blob/910bf41980485c70fef4ef46a3be201c4419dfce/packages/core/src/browser/quick-open/quick-command-service.ts#L166-L174

Also, the wording of the sentence looks odd to me, see if you can improve it.

```ts
// Opens a file for the given URI.
```

Reply:
Updated ;)